src/main.rs

```diff
@@ -1532,6 +1579,20 @@ mod tests {
         );
     }
 
+    #[test]
+    fn compile_cat_end_of_input() {
+        let f = Format::Cat(Box::new(is_byte(0x00)), Box::new(Format::EndOfInput));
```

Comment:
Oh I was thinking this could use Repeat to check that compilation works and the loop knows when to stop